akka-actor/src/main/scala/akka/io/InetAddressDnsResolver.scala

```diff
@@ -1,34 +1,68 @@
+/**
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
 package akka.io
 
-import java.net.{ UnknownHostException, InetAddress }
+import java.net.{ InetAddress, UnknownHostException }
+import java.security.Security
 import java.util.concurrent.TimeUnit
 
 import akka.actor.Actor
 import com.typesafe.config.Config
 
 import scala.collection.immutable
-import sun.net.{ InetAddressCachePolicy ⇒ IACP }
 import akka.util.Helpers.Requiring
 
+import scala.util.Try
+
+/** Respects the settings that can be set on the Java runtime via parameters. */
 class InetAddressDnsResolver(cache: SimpleDnsCache, config: Config) extends Actor {
 
-  import IACP.NEVER // 0 constant
+  // Controls the cache policy for successful lookups only
+  private final val CachePolicyProp = "networkaddress.cache.ttl"
+  @deprecated("Deprecated property key, keeping for legacy compatibility; replaced by CachePolicyProp") 
+  private final val CachePolicyPropFallback = "sun.net.inetaddr.ttl"
```

Comment:
Deprecate something that is private and just was added? O.o
Just leave a comment instead?

Reply:
Yeah it's the "legacy jvm options", no one should touch them basically :P I can add a comment instead